GitHub Like Diff

sotarok opened the pull request:
Thank you so much @kaz @b4b4r07  

## Spec

- Diff like GitHub
- Show diff even if it is the first revision

![image](https://cloud.githubusercontent.com/assets/29064/22617594/dc3af62e-ea7d-11e6-9a67-eeec9b0a8961.png)

## Also

- Close #170 
- Close #171 
- Close #173

suzuki commented:
I found different points from the behavior of the past version.

- "View diff" link
    - Can't toggle
    - The icon did not change
- 1st revision
    - Always open

![_2017-02-08_08_09_28](https://cloud.githubusercontent.com/assets/10488/22716026/27afd26c-edd7-11e6-8546-ec32b7556064.png)

suzuki commented on resource/css/crowi.scss:
```diff
@@ -251,6 +251,11 @@ footer {
   border-radius: 3px;
 }
 
+// adust
```
typo ?

sotarok replied:
:stuck_out_tongue_closed_eyes:

suzuki commented on resource/css/crowi.scss:
```diff
@@ -251,6 +251,11 @@ footer {
 }
 
+// adust
+.d2h-file-header {
```
[memo] class for "diff2html", maybe.

suzuki commented on resource/js/app.js:
```diff
@@ -21,10 +22,13 @@ crowi.fetchUsers();
 const crowiRenderer = new CrowiRenderer();
 window.crowiRenderer = crowiRenderer;
 
+const pageId = $('#content-main').data('page-id');
```
We can use `document.querySelector('#content-main')`
see: https://github.com/crowi/crowi/pull/174/files#diff-39992ee08af86834f2b5095ded60c098R25

suzuki commented on resource/js/app.js:
```diff
@@ -35,3 +39,8 @@ Object.keys(componentMappings).forEach((key) => {
 });
+
+// うわーもうー
+$('a[data-toggle="tab"][href="#revision-history"]').on('show.bs.tab', function() {
```
We can use `document.querySelector('a[data-toggle="tab"][href="#revision-history"]')`.
But I did not test this event using it.

suzuki commented on resource/js/components/Common/Icon.js:
```diff
@@ -0,0 +1,24 @@
+}
+
+Icon.propTypes = {
+  name: React.PropTypes.string,
```
We can use `isRequired` here.
I think the Font Awesome requires name.

suzuki commented on resource/js/components/PageHistory.js:
```diff
@@ -0,0 +1,142 @@
+  constructor(props) {
+    super(props);
+
+    this.crowi = window.crowi; // FIXME
```
How about changing `crowi` to `props`?
see: https://github.com/crowi/crowi/pull/174/files#diff-39992ee08af86834f2b5095ded60c098R37

sotarok replied:
Ah, nice

suzuki commented on resource/js/components/PageHistory.js:
```diff
@@ -0,0 +1,142 @@
+  }
+
+  onDiffOpenClicked(revision)
+  {
```
We need to select (decide) the coding style.

suzuki commented on resource/js/components/PageHistory/Revision.js:
```diff
@@ -0,0 +1,59 @@
+          <strong>{author.username}</strong>
+        </div>
+        <div className="revision-history-meta">
+         <p>
```
indent

sotarok commented:
> "View diff" link

I changed the spec.
To simplize the behavior, i removed the feature of closing diff.
On my mine, 
(It's like "Expand codes around the diff" of GItHub's diff. We can't close the opened diff.)

![image](https://cloud.githubusercontent.com/assets/29064/22718306/7cf74d5a-ed54-11e6-8c33-7e29b9a74e26.png)



> 1st  revision

I also changed the spec.
Because, the first virsion, in my experience, I used to read the original contents of the page.
